tests/test_agent.py

```diff
@@ -138,6 +139,58 @@ def test_maintenance():
     Agent.maintenance_leave()
 
 
+def test_ensure_lock_contention_returns_ex_tempfail(
```

Comment:
two things to validate:

1. does this test, when run in isolation, cause coverage to show that the relevant code was excuted?
2. does the test suite fail if we undo the actual change, but keep this test?